internal/internal_workflow_testsuite_test.go

```diff
@@ -1274,6 +1274,228 @@ func (s *WorkflowTestSuiteUnitTest) Test_GetVersion() {
+		}
+		err := f.Get(ctx, nil) // wait for result
+		if err != nil {
+			return err
+		}
+
+		// test no search attributes
+		wfInfo := GetWorkflowInfo(ctx)
+		s.Nil(wfInfo.SearchAttributes)
+		return err
+	}
+
+	env := s.NewTestWorkflowEnvironment()
+	env.RegisterWorkflow(workflowFn)
+	env.RegisterActivity(oldActivity)
+	env.RegisterActivity(newActivity)
+	env.OnActivity(oldActivity, mock.Anything, "ols_msg").Return("hello_ols_msg", nil).Once()
+	env.ExecuteWorkflow(workflowFn)
+
+	s.True(env.IsWorkflowCompleted())
+	s.Nil(env.GetWorkflowError())
+	env.AssertExpectations(s.T())
+}
+
+func (s *WorkflowTestSuiteUnitTest) Test_GetVersion_ExecuteWithVersion() {
+	oldActivity := func(ctx context.Context, msg string) (string, error) {
+		return "hello" + "_" + msg, nil
+	}
+	newActivity := func(ctx context.Context, msg string) (string, error) {
+		return "hello" + "_" + msg, nil
+	}
+	workflowFn := func(ctx Context) error {
+		ctx = WithActivityOptions(ctx, s.activityOptions)
+		var f Future
+		v := GetVersion(ctx, "test_change_id", DefaultVersion, 2, ExecuteWithVersion(1))
+		if v == DefaultVersion {
+			f = ExecuteActivity(ctx, oldActivity, "ols_msg")
+		} else {
+			f = ExecuteActivity(ctx, newActivity, "new_msg")
+		}
+		err := f.Get(ctx, nil) // wait for result
+		if err != nil {
+			return err
+		}
+
+		// test searchable change version
+		wfInfo := GetWorkflowInfo(ctx)
+		s.NotNil(wfInfo.SearchAttributes)
+		changeVersionsBytes, ok := wfInfo.SearchAttributes.IndexedFields[CadenceChangeVersion]
+		s.True(ok)
+		var changeVersions []string
+		err = json.Unmarshal(changeVersionsBytes, &changeVersions)
+		s.NoError(err)
+		s.Equal(1, len(changeVersions))
+		s.Equal("test_change_id-1", changeVersions[0])
+
+		return err
+	}
+
+	env := s.NewTestWorkflowEnvironment()
+	env.RegisterWorkflow(workflowFn)
+	env.RegisterActivity(oldActivity)
+	env.RegisterActivity(newActivity)
+	env.OnActivity(newActivity, mock.Anything, "new_msg").Return("hello new_mock_msg", nil).Once()
+	env.ExecuteWorkflow(workflowFn)
+
+	s.True(env.IsWorkflowCompleted())
+	s.Nil(env.GetWorkflowError())
+	env.AssertExpectations(s.T())
+}
+
+func (s *WorkflowTestSuiteUnitTest) Test_MockGetVersion_ExecuteWithVersion() {
+	oldActivity := func(ctx context.Context, msg string) (string, error) {
+		return "hello" + "_" + msg, nil
+	}
+	newActivity := func(ctx context.Context, msg string) (string, error) {
+		return "hello" + "_" + msg, nil
+	}
+	workflowFn := func(ctx Context) (string, error) {
+		ctx = WithActivityOptions(ctx, s.activityOptions)
+		var f Future
+		v1 := GetVersion(ctx, "change_1", DefaultVersion, 2, ExecuteWithVersion(1))
+		if v1 == DefaultVersion {
+			f = ExecuteActivity(ctx, oldActivity, "old1")
+		} else {
+			f = ExecuteActivity(ctx, newActivity, "new1")
+		}
+		var ret1 string
+		err := f.Get(ctx, &ret1) // wait for result
+		if err != nil {
+			return "", err
+		}
+
+		v2 := GetVersion(ctx, "change_2", DefaultVersion, 2)
```

Comment:
since the only difference between this test and the next is
```
v1 := GetVersion(ctx, "change_1", DefaultVersion, 2, ExecuteWithVersion(1))
v1 := GetVersion(ctx, "change_1", DefaultVersion, 2, 2)

v2 := GetVersion(ctx, "change_2", DefaultVersion, 2)
v2 := GetVersion(ctx, "change_2", DefaultVersion, 2, ExecuteWithMinVersion())
```
... I kinda think this isn't testing much related to the new code.  some things about it, but mostly just "if mock-versioned, it still uses that version".

probably deserves a "when not mocking GetVersion, the test suite behaves like prod, and respects the WithVersion/MinVersion value" test too.  which can probably just be one, checking that both activities are not using the max values.

Reply:
Yeah, the test ensures that despite on the options, `OnGetVersion` mocks behavior. 

> probably deserves a "when not mocking GetVersion, the test suite behaves like prod, and respects the WithVersion/MinVersion value" test too. 

There are already tests: `Test_GetVersion_ExecuteWithVersion` and `Test_GetVersion_ExecuteWithMinVersion` there are no mocking `GetVersion`